router/router.go

```diff
@@ -141,8 +152,7 @@ func (r *routeSet) ProcessRoute(uri *url.URL) (newurl string, requirePermission
 }
 
 func (r *routeSet) GetProxy(req *ms_http.Request) (*url.URL, error) {
-	route := r.match(req.URL)
-	if route == nil || !route.target.useProxy {
+	if !r.targetHash[req.URL.Host].useProxy {
```

Comment:
How sure are we that r.targetHash[req.URL.Host] will always be non-nil? I *think* that one can reason that, at present, it will always be non-nil, but in a case like this where it's hard to pin down, it's better to check for nil-ness. Perhaps emit an error message to the log if you do find a nil entry here (so it's like a soft assert). However, I don't think the application should crash if that happens.